src/Microsoft.Diagnostic.Repl/Command/CommandProcessor.cs

```diff
@@ -242,12 +290,67 @@ private void SetProperties(InvocationContext context, object instance)
                     property.Property.SetValue(instance, value);
                 }
 
-                if (_argument != null)
+                if (context != null && _argument != null)
                 {
-                    object value = context.ParseResult.CommandResult.GetValueOrDefault();
+                    object value = null;
+                    ArgumentResult result = context.ParseResult.CommandResult.ArgumentResult;
+                    switch (result)
+                    {
+                        case SuccessfulArgumentResult successful:
+                            value = successful.Value;
+                            break;
+                        case FailedArgumentResult failed:
+                            throw new InvalidOperationException(failed.ErrorMessage);
+                    }
                     _argument.SetValue(instance, value);
                 }
             }
+
+            private object[] BuildArguments(MethodBase methodBase, InvocationContext context)
+            {
+                ParameterInfo[] parameters = methodBase.GetParameters();
+                object[] arguments = new object[parameters.Length];
+                for (int i = 0; i < parameters.Length; i++) {
+                    Type parameterType = parameters[i].ParameterType;
+                    TryGetService(parameterType, context, out arguments[i]);
```

Comment:
What happens for the case this returns false?

Reply:
I ignore false and the argument to the function being invoke will be passed as null.  This allows "optional" services leaving it up the function to check for null.

I'll add a comment to clear that up.